cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/AndTrigger.java

```diff
@@ -46,4 +49,33 @@ public boolean isSatisfied(ProgramSchedule schedule, List<Notification> notifica
   public List<TriggerInfo> getTriggerInfos(TriggerInfoContext context) {
     return getUnitTriggerInfosAddRuntimeArgs(context);
   }
+
+  @Nullable
+  @Override
+  public Trigger updateTriggerWithDeletedProgram(ProgramId programId) {
+    List<SatisfiableTrigger> updatedTriggers = new ArrayList<>();
+    for (Trigger trigger : getTriggers()) {
+      if (trigger instanceof ProgramStatusTrigger &&
+        programId.equals(((ProgramStatusTrigger) trigger).getProgramId())) {
+        // this program status trigger will never be satisfied, so the current AND trigger will never be satisfied
+        return null;
+      }
+      if (trigger instanceof co.cask.cdap.internal.app.runtime.schedule.trigger.AbstractCompositeTrigger) {
```

Comment:
since `AndTrigger` extends `AbstractCompositeTrigger`, which extends `ProtoTrigger.AbstractCompositeTrigger`, it has to use the long name to distinguish `AbstractCompositeTrigger` and `ProtoTrigger.AbstractCompositeTrigger`

Reply:
can you just rename the internal one to CompositeTrigger? This seems silly.